### DISCUSSION TITLE
How to output HTML directly?

### DISCUSSION THREAD

**Discussion by edu-de** [sourcelair/xterm.js#4572] [Q&A] [answered]
I need to print command results which I already have as HTML code.
For example, a "ls" command in my example returns a "< table >...< /table >" string with the "ls" results.

Using term.write("< table >..< /table >") prints the HTML code instead of interpreting it and showing the table instead.
Any idea how to achieve this?

**Comment by jerch** [comment, marked as answer]
Thats not possible in a direct fashion. The terminal works the other way around - it gives you text/cursor mechanics to be shown by  a HTML/JS engine. But to get there, it uses very different render methods depending on the selected renderer (canvas, webgl or DOM), thus it is almost impossible to hook predefined HTML snippets directly into the generated output.

The best you can do here is to parse the content back into a text string as generated by `ls` and then feed it to the terminal, e.g.
```javascript
term.write(yourTableParse('<table>...</table>'));
```

**Reply by edu-de** [reply]
Hi @jerch 
thanks for your answer.
What is the best way to format the term.write() as a table? I don't seem to find any table example.
How to output margins, spaces (centering text), etc ? What would you recommend for "emulating" the html table as output understood by xterm.js ?

**Reply by jerch** [reply]
There are some table generators for the terminal in npm, maybe you can use those or code parts for it? Like [table](https://www.npmjs.com/package/table) or [cli-table](https://www.npmjs.com/package/cli-table)...
